Rules.py

```diff
@@ -84,49 +83,70 @@ def global_rules(world):
 
     ### SOUTH CLOCK TOWN
     set_rule(world.get_location('Clock Town Business Scrub'), lambda state: state.has('Moons Tear'))
-    set_rule(world.get_location('Clock Tower Platform HP'), lambda state: state.form('Human') or state.form('Goron') or state.form('Zora') or (state.form('Deku') and state.can_reach(world.get_location('Clock Town Business Scrub'))))
+    set_rule(world.get_location('Clock Tower Platform HP'), lambda state: state.form('Human') or state.form('Goron') or state.form('Zora') or (state.form('Deku') and state.has('Moons Tear')))
     # @Vlix
     # oh dang, goron link can't just hop up that small ledge? or maybe he doesn't even fit on it, heh
     # gotta test to see ways of getting on there todo: find ways of getting up to that platform in various forms
     # ah, just thought of adding deku form as I was reading through, nice lol
 
-    set_rule(world.get_location('Festival Tower Rupee Chest'), lambda state: (state.has('Hookshot') and state.form('Human')) or (state.form('Deku') and state.can_reach(world.get_location('Clock Town Business Scrub'))))
-    set_rule(world.get_location('South Clock Town Hookshot Ledge Rupee Chest'), lambda state: state.has('Hookshot') and state.form('Human'))
+    set_rule(world.get_location('Festival Tower Rupee Chest'), lambda state: state.can_use('Hookshot') or (state.form('Deku') and state.can_reach(world.get_location('Clock Town Business Scrub'))))
+    set_rule(world.get_location('SCT 20 Rupee Chest'), lambda state: state.can_use('Hookshot'))
 
-    # set_rule(world.get_location('Ocarina of Time'), lambda state: (state.form('Deku') and state.has('Magic Meter')) or (state.has('Bow') and state.form('Human')) or state.form('Zora'))
     set_rule(world.get_location('Ocarina of Time'), lambda state: state.can_pop_balloon())
     # is this right? it looks like the check is just to hit skull kid in the air, functionally the same as popping a balloon lol
+    set_rule(world.get_entrance('End of First Cycle'), lambda state: state.has('Ocarina'))
```

Comment:
technically this should require the song of time; this is meant to be a gate from the first cycle into the main game. what I'm thinking, as a way of requiring that the song of time is placed in the first cycle, is to essentially have copies of all the clock town regions, and have the 2nd+ cycle ones require that the first cycle be done (which might just mean requiring song of time)

so this is an 'entrance' because it would lead from the 1st cycle clock tower roof to the 2nd+ cycle south clock town